.github/workflows/ci.yaml

```diff
@@ -179,13 +184,15 @@ jobs:
         run: sudo rm -rf /usr/local/lib/android /usr/share/dotnet /opt/ghc /opt/hostedtoolcache/CodeQL && sudo docker image prune --all --force
       - name: checkout source code
         uses: actions/checkout@v4
-      - name: Install tox
-        run: sudo apt update && sudo apt install tox
       - name: Set up Python
         uses: actions/setup-python@v5
         with:
           python-version: "3.x"
-
+      - name: Install tox
+        run: |
+          python3 --version
+          python3 -m ensurepip
+          python3 -m pip install tox
```

Comment:
This only installs tox for the user but not for the superuser and hence line 393 with `sudo ... python3 -m tox …` fails

Reply:
Yes, .. this is a draft, it doesn't require review/request changes.